frontend/app/routes/protected/person-case/secondary-doc.tsx

```diff
@@ -0,0 +1,213 @@
+
+  return {
+    documentTitle: t('protected:secondary-identity-document.page-title'),
+    defaultFormValues: context.session.inPersonSINCase?.secondaryDocument,
+  };
+}
+
+export function meta({ data }: Route.MetaArgs) {
+  return [{ title: data.documentTitle }];
+}
+
+export async function action({ context, request }: Route.ActionArgs) {
+  requireAuth(context.session, new URL(request.url), ['user']);
+  const lang = getLanguage(request);
+  const t = await getFixedT(request, handle.i18nNamespace);
+
+  const formData = await request.formData();
+  const action = formData.get('action');
+  const maxImageSizeBits = 1024 * 1024 * 15; //Max image size is 15 MB
+
+  switch (action) {
+    case 'back': {
+      throw i18nRedirect('routes/protected/person-case/primary-docs.tsx', request);
+    }
+    case 'next': {
+      const dateFormatEST = new Intl.DateTimeFormat('en-CA', {
+        timeZone: 'America/Toronto',
+        year: 'numeric',
+        month: '2-digit',
+        day: '2-digit',
+      });
+
+      //Normalized date at current day midnight EST
+      const currentDate = new Date(`${dateFormatEST.format(new Date())}T00:00:00`);
+
+      const schema = v.object({
+        documentType: v.picklist(VALID_DOCTYPE, t('protected:secondary-identity-document.document-type.invalid')),
+        document: v.pipe(
+          v.file(t('protected:secondary-identity-document.upload-document.required')),
+          v.mimeType(
+            ['image/jpeg', 'image/png', 'image/heic'],
+            t('protected:secondary-identity-document.upload-document.invalid'),
+          ),
+          v.maxSize(maxImageSizeBits),
+        ),
+        expiryYear: v.pipe(
+          v.number(t('protected:secondary-identity-document.expiry-date.required-year')),
+          v.integer(t('protected:secondary-identity-document.expiry-date.invalid-year')),
+          v.minValue(currentDate.getFullYear(), t('protected:secondary-identity-document.expiry-date.invalid-year')),
+        ),
+        expiryMonth: v.pipe(
+          v.number(t('protected:secondary-identity-document.expiry-date.required-month')),
+          v.integer(t('protected:secondary-identity-document.expiry-date.invalid-month')),
+          v.minValue(1, t('protected:secondary-identity-document.expiry-date.invalid-month')),
+          v.maxValue(12, t('protected:secondary-identity-document.expiry-date.invalid-month')),
+        ),
+        expiryDay: v.pipe(
+          v.number(t('protected:secondary-identity-document.expiry-date.required-day')),
+          v.integer(t('protected:secondary-identity-document.expiry-date.invalid-day')),
+          v.minValue(1, t('protected:secondary-identity-document.expiry-date.invalid-day')),
+          v.maxValue(31, t('protected:secondary-identity-document.expiry-date.invalid-day')),
+        ),
+        expiryDate: v.pipe(v.date(), v.minValue(currentDate, t('protected:secondary-identity-document.expiry-date.invalid'))),
```

Comment:
This should be updated to use the functions from `utils/date-utils.ts`. You should do something like this: (pseudocode)

``` ts
isBefore(currentDate, getStartOfDayInTimezone(serverEnvironment.BASE_TIMEZONE, expiryDate))
```